deps/key-value-storage/src/lib.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A key-value storage interface and implementations.
+
+use async_trait::async_trait;
+use serde::Deserialize;
+
+use std::{sync::Arc, time::Duration};
+
+pub mod error;
+pub use error::{KeyValueStorageError, Result};
+
+pub mod simple;
+
+#[cfg(feature = "postgres")]
+pub mod postgres;
+
+#[derive(Default)]
+pub struct SetContext {
+    /// Whether to overwrite the existing value.
+    pub overwrite: bool,
+
+    /// The TTL of the value.
+    /// If the TTL is set, the value will be deleted after the TTL expires.
+    /// If the TTL is not set, the value will not be deleted.
+    /// Note that not all implementations support TTL.
+    pub ttl: Option<Duration>,
+}
+
+#[async_trait]
+pub trait KeyValueStorage: Send + Sync {
+    /// Set a value for a key.
+    async fn set(&self, key: String, value: Vec<u8>, context: SetContext) -> Result<()>;
+
+    /// List all keys.
+    async fn list(&self) -> Result<Vec<String>>;
+
+    /// Get a value for a key.
+    async fn get(&self, key: String) -> Result<Option<Vec<u8>>>;
+
+    /// Delete a value for a key.
+    /// Return the deleted value if it exists.
+    async fn delete(&self, key: String) -> Result<Option<Vec<u8>>>;
+
+    /// Check if the key is valid.
+    ///
+    /// The key is valid if it only contains ASCII alphanumeric characters, `-` or `_`.
+    /// No spaces and other special characters are allowed to prevent SQL injection.
```

Comment:
I think it's reasonable to restrict the valid key chars (we want them to be representable as url path segments). However, if we want to use this store for kbs secrets, this restriction will be a breaking change probably, since keys like `luks.key` are allowed, no?

In any case this is not about SQL injection, we don't string-template keys into the SQL statements.

Reply:
Ok. Let me move it out and add `.` as a valid charactor.